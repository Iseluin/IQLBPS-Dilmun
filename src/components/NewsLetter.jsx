import React from 'react'

const NewsLetter = () => {
  return (
    <div className='m-14 p-4 md:flex rounded-md shadow-md justify-around bg-primary'>
      <div className='text-white '>
        <h1 className='text-[30px] sm:text-[40px] mb-3 sm:mb-0'>Subscribe to our <br /> NEWSLETTER</h1>
      </div>
      <div className='items-center sm:flex justify-center gap-2'>
        <input type="text" placeholder='Write your email here' className='rounded p-2 mb-3 sm:mb-0 placeholder:text-primary border border-solid ' />
        <button type='submit' className='p-2 rounded text-primary bg-white'>Submit</button>
      </div>
    </div>

  )
}

export default NewsLetter