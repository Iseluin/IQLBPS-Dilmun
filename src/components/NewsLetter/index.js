import { addDoc, collection, getDocs, query, where } from 'firebase/firestore';
import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { Trans, useTranslation } from 'react-i18next';
import { db } from '../../firebase-config';
import CustomButton from '../CustomComponents/CustomButton';

const NewsLetter = () => {
  const { t } = useTranslation();
  const [error, setError] = useState('');
  const { register, handleSubmit, reset } = useForm();

  const onSubmit = async ({ email }) => {
    const subscriberRef = collection(db, 'Subscribers');
    const q = query(subscriberRef, where('email', '==', email));
    const querySnapshot = await getDocs(q);
    if (querySnapshot.empty) {
      await addDoc(subscriberRef, {
        email,
      });
      setError(null);
      reset();
    } else {
      setError(t('error.exist_email'));
    }
  };
  return (
    <section data-testid="news-letter-section">
      <div className="pt-14 pb-14 pr-10 pl-10 md:flex shadow-md justify-around bg-primary">
        <div className="text-background flex flex-col md:flex-row justify-center items-center">
          <img
            className="w-24 md:w-40  h-100"
            src="https://www.freepnglogos.com/uploads/blue-email-logo-png-4.png"
            alt="email"
          />
          <h1 className="text-[20px] sm:text-[40px] mb-3 sm:mb-0">
            <Trans
              i18nKey="news_letter.subscribe_to_our_newsletter"
              components={{ br: <br /> }}
            />
          </h1>
        </div>
        <div className="mt-4">
          <p className="text-background mb-5">{t('news_letter.description')}</p>
          <div className="items-center">
            <form onSubmit={handleSubmit(onSubmit)}>
              <input
                type="text"
                placeholder={t('news_letter.email')}
                name="email"
                {...register('email')}
                className="rounded p-1 w-full sm:w-fit md:p-2 mb-3 mr-1 sm:mb-0 placeholder:text-background border border-solid "
                required
              />
              <CustomButton paddingX="px-2" paddingY="py-2">
                {t('news_letter.subscribe')}
              </CustomButton>
            </form>
          </div>
          {error && <p className="text-red-500 mt-1">{error}</p>}
        </div>
      </div>
    </section>
  );
};

export default NewsLetter;
