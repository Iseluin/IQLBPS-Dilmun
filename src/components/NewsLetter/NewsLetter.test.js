import React from "react";
import { render } from "@testing-library/react";
import NewsLetter from ".";


it('render correctly', () => {
    const { queryByPlaceholderText, getByText, queryByAltText} = render(<NewsLetter />)

    expect(queryByPlaceholderText('Write your email here')).toBeTruthy()
    expect(getByText('Subscribe to our newsletter and stay updated')).toBeTruthy()
    expect(queryByAltText('email')).toBeTruthy()
    expect(getByText("Subscribe to our Newsletter !")).toMatchSnapshot(`
    <h1 className='text-[20px] sm:text-[40px] mb-3 sm:mb-0'>Subscribe to our <br /> Newsletter !</h1>
  `);
})