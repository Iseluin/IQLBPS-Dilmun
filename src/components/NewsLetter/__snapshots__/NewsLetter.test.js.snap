// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<section
  data-testid="news-letter-section"
>
  <div
    class="pt-14 pb-14 pr-10 pl-10 md:flex shadow-md justify-around bg-primary"
  >
    <div
      class="text-background flex flex-col md:flex-row justify-center items-center"
    >
      <img
        alt="email"
        class="w-24 md:w-40  h-100"
        src="https://www.freepnglogos.com/uploads/blue-email-logo-png-4.png"
      />
      <h1
        class="text-[20px] sm:text-[40px] mb-3 sm:mb-0"
      />
    </div>
    <div
      class="mt-4"
    >
      <p
        class="text-background mb-5"
      >
        news_letter.description
      </p>
      <div
        class="items-center"
      >
        <form>
          <input
            class="rounded p-1 w-full sm:w-fit md:p-2 mb-3 mr-1 sm:mb-0 placeholder:text-background border border-solid "
            name="email"
            placeholder="news_letter.email"
            required=""
            type="text"
          />
          <button
            class="bg-tertiary px-2 py-2 w-full sm:w-auto  text-white font-semibold rounded-lg hover:shadow-lg hover:bg-secondary transition duration-500 cursor-pointer"
            type="submit"
          >
            news_letter.subscribe
          </button>
        </form>
      </div>
    </div>
  </div>
</section>
`;
